tests/tools/triage/ticket_search/ticket_fetcher_test.py

```diff
@@ -0,0 +1,91 @@
+from unittest.mock import MagicMock
+from tools.triage.ticket_search.ticket_fetcher import TicketFetcher
+from tools.triage.ticket_search.triage_ticket import TriageTicket
+
+def setup_jira_client():
+    jira_client = MagicMock()
+    return jira_client
+
+def setup_ticket_parser():
+    ticket_parser = MagicMock()
+    return ticket_parser
+
+def setup_directory_cache_fetcher():
+    directory_cache_checker = MagicMock()
+    return directory_cache_checker
+
+def setup_ticket_fetcher(ticket_parser, jira_client, directory_cache_checker):
+    return TicketFetcher(ticket_parser, jira_client, directory_cache_checker)
+
+def setup_mock_issue(issue_id, component):
+    """Set up a mock issue"""
+    issue = MagicMock()
+    issue.key = issue_id
+    issue.fields = MagicMock()
+    issue.fields.components = [MagicMock()]
+    issue.fields.components[0].name = component
+    issue.fields.description = f"Description for {issue_id}"
+    return issue
+
+def setup_ticket_query(jql):
+    query = MagicMock()
+    query.build = MagicMock(return_value=jql)
+    return query
+
+def test_ticket_should_be_parsed_if_it_is_a_triage_issue():
+    """If the ticket is a triage issue then we will parse it"""
+    jira_client = setup_jira_client()
+    ticket_parser = setup_ticket_parser()
+    directory_cache_checker = setup_directory_cache_fetcher()
+    ticket_fetcher = setup_ticket_fetcher(ticket_parser, jira_client, directory_cache_checker)
+    issue_id = "MGMT-12345"
+    mock_issue = setup_mock_issue(issue_id, "Cloud-Triage")
+    jira_client.issue = MagicMock(return_value=mock_issue)
+    ticket_fetcher.fetch_single_ticket_by_key(issue_id)
+    jira_client.issue.assert_called_once_with(issue_id)
+    ticket_parser.parse.assert_called_once_with(mock_issue)
+
+def test_ticket_should_not_be_parsed_if_it_is_not_a_triage_issue():
+    """If the ticket is not a triage issue then we will skip it"""
+    jira_client = setup_jira_client()
+    ticket_parser = setup_ticket_parser()
+    directory_cache_checker = setup_directory_cache_fetcher()
+    ticket_fetcher = setup_ticket_fetcher(ticket_parser, jira_client, directory_cache_checker)
+    issue_id = "MGMT-12345"
+    mock_issue = setup_mock_issue(issue_id, "Some-other-component")
+    jira_client.issue = MagicMock(return_value=mock_issue)
+    ticket_fetcher.fetch_single_ticket_by_key(issue_id)
+    jira_client.issue.assert_called_once_with(issue_id)
+    ticket_parser.parse.assert_not_called()
+
+def test_fetch_page_by_query():
+    """Test fetch by query"""
+    jira_client = setup_jira_client()
+    ticket_parser = setup_ticket_parser()
+    directory_cache_checker = setup_directory_cache_fetcher()
+    ticket_fetcher = setup_ticket_fetcher(ticket_parser, jira_client, directory_cache_checker)
```

Comment:
Any reason why we are not using setup methods? It would definitely reduce the boilerplate code